src/notebook.c

```diff
@@ -65,6 +66,8 @@ static gboolean switch_in_progress = FALSE;
 static GtkWidget *switch_dialog = NULL;
 static GtkWidget *switch_dialog_label = NULL;
 
+static gboolean doc_saves_to_new_file = FALSE;
```

Comment:
> Ideally there wouldn't be any need for a hack to know whether it was saved or saved as

Would that unexpectedly move a manually rearranged tab when simply just  saving a file?

> it only finds out whether it's the first save, not whether the filename changed.

As I observed, `doc->real_path` is always `NULL` in a `"document-before-save"` event when saving to a new file.  Is there a case where this is different?  I'll check it again.